core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks.go

```diff
@@ -0,0 +1,828 @@
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    if len(assetAgreement.Recipient) == 0 {
+        errorMsg = "Empty lock recipient"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    if len(lockInfoHTLC.Hash) == 0 {
+        errorMsg = "Empty lock hash value"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    assetAgreementBytes, err := proto.Marshal(assetAgreement)
+    if err != nil {
+        log.Error(err.Error())
+        return false, err
+    }
+    lockInfoBytes, err := proto.Marshal(lockInfoHTLC)
+    if err != nil {
+        log.Error(err.Error())
+        return false, err
+    }
+    iccResp := stub.InvokeChaincode(am.interopChaincodeId, [][]byte{[]byte("LockFungibleAsset"), assetAgreementBytes, lockInfoBytes}, "")
+    fmt.Printf("Response from Interop CC: %+v\n", iccResp)
+    if iccResp.GetStatus() != shim.OK {
+        return false, fmt.Errorf(string(iccResp.GetPayload()))
+    }
+    lockExpiryTimeSec := int64(lockInfoHTLC.ExpiryTimeMillis)/1000
+    lockExpiryTimeNano := (int64(lockInfoHTLC.ExpiryTimeMillis) - (lockExpiryTimeSec * 1000)) * (1000 * 1000)     // Convert milliseconds to nanoseconds
+    fmt.Printf("%d units of asset type %s locked for %s until %+v\n", assetAgreement.NumUnits, assetAgreement.Type, assetAgreement.Recipient, time.Unix(lockExpiryTimeSec, lockExpiryTimeNano))
+    return true, nil
+}
+
+// If 'assetAgreement.Locker' or 'assetAgreement.Recipient' is blank, assume it's the caller
+func (am *AssetManagement) IsAssetLocked(stub shim.ChaincodeStubInterface, assetAgreement *common.AssetAgreement) (bool, error) {
+    var infoMsg, errorMsg string
+
+    if len(am.interopChaincodeId) == 0 {
+        errorMsg = "Interoperation chaincode ID not set. Run the 'Configure(...) function first."
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+
+    if len(assetAgreement.Type) == 0 {
+        errorMsg = "Empty asset type"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    if len(assetAgreement.Id) == 0 {
+        errorMsg = "Empty asset ID"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    myselfBytes, err := stub.GetCreator()
+    if err != nil {
+        log.Error(err.Error())
+        return false, err
+    }
+    myself := string(myselfBytes)
+    if len(assetAgreement.Recipient) == 0 {
+        infoMsg = "Empty lock recipient; assuming caller"
+        log.Info(infoMsg)
+        assetAgreement.Recipient = myself
+    }
+    if len(assetAgreement.Locker) == 0 {
+        infoMsg = "Empty locker; assuming caller"
+        log.Info(infoMsg)
+        assetAgreement.Locker = myself
+    }
+    if assetAgreement.Recipient == assetAgreement.Locker {
+        errorMsg = "Invalid query: locker identical to recipient"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
```

Comment:
Should a user not be able to query whether an asset of a specific ID is locked for/or by them, without specifying the counterparty?

Reply:
Yes, they should. I'd envisioned the use of a `*` for this purpose, but that will be interpreted by the Fabric Interop CC. @knarayan please take note.

In fact, following on the previous comment, a user should be able to check whether an asset is locked period, regardless of whether it is a party to the agreement. I'll review the code to determine if all bases are covered.

Reply:
Done in the base class. Fabric Interop CC needs to handle *.